code/modules/unit_tests/monkey_business.dm

```diff
@@ -0,0 +1,59 @@
+/**
+ * Monkey Business
+ *
+ * This unit test spawns a predefined number of monkies, each of which
+ * are set to have a 100% chance of attempting to use something next to them each Life
+ *
+ * This test basically just checks to see if attack procs are working correctly,
+ * but its also hilarious and fun to watch locally.
+ */
+/datum/unit_test/monkey_business
+	priority = PRIORITY_LOW + 1 // should be the last test to run
+	var/monkey_count = 50
+	var/monkey_timer = 30 SECONDS
+	var/start_runtimes = 0
+	var/list/monkey_list
+	var/list/running = TRUE
+
+/datum/unit_test/monkey_business/Run()
+	monkey_list = list()
+	start_runtimes = GLOB.total_runtimes
+	for(var/monkey_id in 1 to monkey_count)
+		var/turf/spawn_turf = get_safe_random_station_turf()
+		var/mob/living/carbon/human/monkey = new /mob/living/carbon/human(spawn_turf)
+		monkey.set_name("Monkey [monkey_id]")
+		monkey.set_species(/datum/species/monkey)
+		if(prob(10)) // BLOOD FOR THE BLOOD GODS
+			monkey.put_in_active_hand(new /obj/item/knife/shiv)
+			new /datum/ai_controller/monkey/angry(monkey)
+		else
+			new /datum/ai_controller/monkey(monkey)
+		monkey_list += monkey
+	addtimer(CALLBACK(src, .proc/finalize), monkey_timer)
+	while(running)
+		sleep(2)
```

Comment:
2 TICKS? 2 Deciseconds? 0.2 SECONDS? How long do you want the sleep to be for? Units are quite good to use to ensure clear communication. we have TICKS and SECONDS defines to help with this.

Reply:
fair point